### PR TITLE
Change button_widget class to btn-primary

### LINKED ISSUE #29114: Would make more sense that form submit buttons have boostrap class btn-primary
**Description**  
Bootstrap template update from 3.3 to 4.x replaced `btn-default` class with `btn-secondary`, which results in pale grey submit buttons.
https://github.com/symfony/symfony/blob/8d277ce3e594c196594170e4fbbc37be98ba4f73/src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig#L148

I believe it would make more sense to give it `btn-primary` class as documented in the form section of Bootstrap 4 documentation: https://getbootstrap.com/docs/4.0/components/forms/

**Example**  

Offical documentation and examples
image

Current template
image

### PR DESCRIPTION
**Pull request by JulienMaille**
FIX #29114

| Q             | A
| ------------- | ---
| Branch?       | master for features / 2.8 up to 4.1 for bug fixes 
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #29114   
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Change button_widget class to btn-primary in Bootstrap 4 twig template

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig [hunk 1]
```diff
@@ -145,7 +145,7 @@
 {%- endblock widget_attributes -%}
 
 {% block button_widget -%}
-    {%- set attr = attr|merge({class: (attr.class|default('btn-secondary') ~ ' btn')|trim}) -%}
+    {%- set attr = attr|merge({class: (attr.class|default('btn-primary') ~ ' btn')|trim}) -%}
```
[the remark this answers is not in the document]
Comment (javiereguiluz) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig]: I agree. Let's make this change ... but only for the `submit` button. @xabbuh can you give us any clue about how to do that only for this specific type of button? Thanks!

[the remark this answers is not in the document]
Comment (arkste) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig]: @javiereguiluz 

you can either check `type == 'submit'` in `button_widget` or overwrite `class` in a `submit_widget` block.

something like ...

```
{%- block submit_widget -%}
    {%- set attr = attr|merge({class: (attr.class|default('btn-primary') ~ ' btn')|trim}) -%}
    {{- parent() -}}
{%- endblock submit_widget -%}
```

... should work, if you revert the change in `button_widget`.

[the remark this answers is not in the document]
Comment (neFAST) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig]: > I guess this should probably not be set for every button but only for submit buttons, shouldn't it?

Well, why would 'secondary' be the new default class for other buttons?

[the remark this answers is not in the document]
Comment (neFAST) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig]: I tried and can confirm it works as intended. Should I close that PR and create a new one?

[the remark this answers is not in the document]
Comment (neFAST) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig]: @arkste Should we use
`{%- endblock submit_widget -%}`
or
`{%- endblock submit_widget %}`
most of the blocks end without dash (but not all of them)

[the remark this answers is not in the document]
Comment (stof) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig]: trimming whitespaces after the `{% endblock %}` tag is useless, as the form theme won't render the spaces being between top-level blocks (as it renders blocks)

[the remark this answers is not in the document]
Comment (xabbuh) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig]: @neFAST You can just push additional commits to the branch this PR is based on to update it.

[the remark this answers is not in the document]
Comment (neFAST) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig]: @xabbuh Done

**Review by xabbuh — DISMISSED** [review]
As a bugfix for 3.4?

**Comment by xabbuh** [comment]
@neFAST Can you take a look at the failing tests?

**Comment by neFAST** [comment]
@xabbuh fixed

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
@neFAST Thanks for your fast reaction. Nice contribution. 👍 IMO this is a bugfix so I have updated the description accordingly. Don't worry about the target branch. We can rebase your changes on the right branch while merging.

**Review by dunglas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @neFAST.

**Pull request merged**